be/src/exec/hash_join_node_ir.cpp

```diff
@@ -137,11 +137,20 @@ int HashJoinNode::process_probe_batch(RowBatch* out_batch, RowBatch* probe_batch
     return rows_returned;
 }
 
+// when build table has too many duplicated rows, the collisions will be very serious,
+// so in some case will don't need to store duplicated value in hash table, we can build an unique one
 void HashJoinNode::process_build_batch(RowBatch* build_batch) {
     // insert build row into our hash table
     for (int i = 0; i < build_batch->num_rows(); ++i) {
-        _hash_tbl->insert(build_batch->get_row(i));
+        if (_join_op == TJoinOp::LEFT_ANTI_JOIN|| _join_op == TJoinOp::RIGHT_ANTI_JOIN
```

Comment:
I think these compare can be done just once when initializing the HashJoinNode, no need to judge it for every row.

Reply:
done